src/core/router/identity.cc

```diff
@@ -729,42 +727,47 @@ Keys CreateRandomKeys() {
 
 IdentHash CreateRoutingKey(
     const IdentHash& ident) {
-  std::uint8_t buf[41];  // ident + yyyymmdd
-  memcpy(buf, (const std::uint8_t *)ident, 32);
-  time_t t = time(nullptr);
-  struct tm tm;
-  // TODO(unassigned): never use sprintf, use snprintf instead.
-#ifdef _WIN32
-  gmtime_s(&tm, &t);
-  sprintf_s(
-      reinterpret_cast<char *>((buf + 32)),
-      9,
-      "%04i%02i%02i",
-      tm.tm_year + 1900,
-      tm.tm_mon + 1,
-      tm.tm_mday);
-#else
-  gmtime_r(&t, &tm);
-  sprintf(
-      reinterpret_cast<char *>((buf + 32)),
-      "%04i%02i%02i",
-      tm.tm_year + 1900,
-      tm.tm_mon + 1,
-      tm.tm_mday);
-#endif
+  if (ident.IsZero())
+    {
+      throw std::invalid_argument(
+          "Identity: cannot create routing key from zero-initialized ident");
+    }
   IdentHash key;
-  // TODO(anonimal): this try block should be larger or handled entirely by caller
-  try {
-    kovri::core::SHA256().CalculateDigest((std::uint8_t *)key, buf, 40);
-  } catch (...) {
-    core::Exception ex;
-    ex.Dispatch(__func__);
-    // TODO(anonimal): review if we need to safely break control, ensure exception handling by callers
-    throw;
-  }
+  try
```

Comment:
Extend this try block to:

```diff
diff --git a/src/core/router/identity.cc b/src/core/router/identity.cc
index ccd6c862..ed793bc2 100644
--- a/src/core/router/identity.cc
+++ b/src/core/router/identity.cc
@@ -725,16 +725,17 @@ Keys CreateRandomKeys() {
   return keys;
 }
 
-IdentHash CreateRoutingKey(
-    const IdentHash& ident) {
-  if (ident.IsZero())
-    {
-      throw std::invalid_argument(
-          "Identity: cannot create routing key from zero-initialized ident");
-    }
+IdentHash CreateRoutingKey(const IdentHash& ident)
+{
   IdentHash key;
   try
     {
+      if (ident.IsZero())
+        {
+          throw std::invalid_argument(
+              "Identity: cannot create routing key from zero-initialized "
+              "ident");
+        }
       const std::string formatted_key = FormatRoutingKey(ident);
       kovri::core::SHA256().CalculateDigest(
           key(),
```

Reply:
I extended the try-block, but Github is still picking it up as unchanged.